operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/informer/InformerConfiguration.java

```diff
@@ -57,15 +70,61 @@ public InformerConfigurationBuilder<R> withSecondaryToPrimaryMapper(
     }
 
     public InformerConfigurationBuilder<R> withNamespaces(String... namespaces) {
-      this.namespaces = namespaces != null ? Set.of(namespaces) : Collections.emptySet();
-      return this;
+      return withNamespaces(
+          namespaces != null ? Set.of(namespaces) : ResourceConfiguration.DEFAULT_NAMESPACES);
```

Comment:
This `DEFAULT_NAMESPACES` is a little confusing. Should we rename this to `WATCH_ALL_NAMESPACES_ON_CLUSTER` or such?

Since it is already defined as:
`Set<String> DEFAULT_NAMESPACES = Collections.singleton(Constants.WATCH_ALL_NAMESPACES);`

see:
https://github.com/java-operator-sdk/java-operator-sdk/blob/b9ee05e6e3c803be0b1e95633f465a632bd1dd52/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ResourceConfiguration.java#L13-L14

@metacosm

Reply:
I used `DEFAULT_NAMESPACES` because I thought it'd be a good idea to decouple the notion of what is used for the default value from the actual value so that it's cleared at the usage site that we're using the default value because that's what should be used at that spot instead of an arbitrary value. 

I do agree that it's a little confusing but I think it's better to keep this semantics to make it explicit that the default value is used at these spots. This way, the code using the value should also not change if the default value were to change.

Reply:
Yes that makes sense, I just I think found some places where it was not used, just the value directly, will rather then create PR for that if I find it again (don't remember where it was)

Reply:
https://github.com/java-operator-sdk/java-operator-sdk/blob/80f631e72d7b0dbd17802435490d8944b72ef742/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/AnnotationControllerConfiguration.java#L76-L76

I think here it would be good to use it too

Reply:
will take a look

Reply:
Yes, that's possible, unfortunately, ran into a couple of such occurrences myself as well and tried to fix them when I could.